baremetal/machine_manager_test.go

```diff
@@ -0,0 +1,1626 @@
+			t.Errorf("%s: unexpected error \"%v\" (expected \"%v\")",
+				tc.CaseName, err, tc.ExpectedResult)
+		}
+		if tc.Host != nil {
+			key := client.ObjectKey{
+				Name:      tc.Host.Name,
+				Namespace: tc.Host.Namespace,
+			}
+			host := bmh.BareMetalHost{}
+
+			if tc.Host != nil {
+				err := c.Get(context.TODO(), key, &host)
+				if err != nil {
+					t.Errorf("%s: Get failed, %v",
+						tc.CaseName, err)
+				}
+			}
+
+			name := ""
+			expectedName := ""
+			if host.Spec.ConsumerRef != nil {
+				name = host.Spec.ConsumerRef.Name
+			}
+			if tc.ExpectedConsumerRef != nil {
+				expectedName = tc.ExpectedConsumerRef.Name
+			}
+			if name != expectedName {
+				t.Errorf("%s: expected ConsumerRef %v, found %v",
+					tc.CaseName, tc.ExpectedConsumerRef, host.Spec.ConsumerRef)
+			}
+		}
+	}
+}
+
+func newConfig(t *testing.T, UserDataNamespace string, labels map[string]string, reqs []capbm.HostSelectorRequirement) (*capbm.BareMetalMachine, *corev1.ObjectReference) {
+	config := capbm.BareMetalMachine{
+		Spec: capbm.BareMetalMachineSpec{
+			Image: capbm.Image{
+				URL:      testImageURL,
+				Checksum: testImageChecksumURL,
+			},
+			UserData: &corev1.SecretReference{
+				Name:      testUserDataSecretName,
+				Namespace: UserDataNamespace,
+			},
+			HostSelector: capbm.HostSelector{
+				MatchLabels:      labels,
+				MatchExpressions: reqs,
+			},
+		},
+	}
+
+	infrastructureRef := &corev1.ObjectReference{
+		Name:       "someothermachine",
+		Namespace:  "myns",
+		Kind:       "Machine",
+		APIVersion: capi.GroupVersion.String(),
+	}
+	return &config, infrastructureRef
+}
+
+func TestUpdateMachineStatus(t *testing.T) {
+	scheme := runtime.NewScheme()
+	err := capi.AddToScheme(scheme)
+
+	if err != nil {
+		log.Printf("AddToScheme failed: %v", err)
+	}
+
+	nic1 := bmh.NIC{
+		IP: "192.168.1.1",
+	}
+
+	nic2 := bmh.NIC{
+		IP: "172.0.20.2",
+	}
+
+	testCases := []struct {
+		Host            *bmh.BareMetalHost
+		Machine         *capi.Machine
+		ExpectedMachine capi.Machine
+		BMMachine       capbm.BareMetalMachine
+	}{
+		{
+			// machine status updated
+			Host: &bmh.BareMetalHost{
+				Status: bmh.BareMetalHostStatus{
+					HardwareDetails: &bmh.HardwareDetails{
+						NIC: []bmh.NIC{nic1, nic2},
+					},
+				},
+			},
+			Machine: &capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mymachine",
+					Namespace: "myns",
+				},
+				Status: capi.MachineStatus{},
+			},
+			ExpectedMachine: capi.Machine{
+				Status: capi.MachineStatus{
+					Addresses: []capi.MachineAddress{
+						capi.MachineAddress{
+							Address: "192.168.1.1",
+							Type:    "InternalIP",
+						},
+						capi.MachineAddress{
+							Address: "172.0.20.2",
+							Type:    "InternalIP",
+						},
+					},
+				},
+			},
+			BMMachine: capbm.BareMetalMachine{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "BMMachine",
+					APIVersion: capi.GroupVersion.String(),
+				},
+			},
+		},
+		{
+			// machine status unchanged
```

Comment:
maybe a changed status with updated ip addresses should be done instead here, otherwise redundant case